sklearn/cross_validation.py

```diff
@@ -1010,14 +1010,19 @@ def _iter_indices(self):
             # Because of rounding issues (as n_train and n_test are not
             # dividers of the number of elements per class), we may end
             # up here with less samples in train and test than asked for.
-            if len(train) < self.n_train or len(test) < self.n_test:
+            if len(train) + len(test) < self.n_train + self.n_test:
```

Comment:
This condition is more correct since you can have `len(test) < self.n_test` but no all indices have already been assigned to either train or test.